packages/connect-ui/src/components/modal/modal.tsx

```diff
@@ -0,0 +1,41 @@
+import { Component, h, Prop, Event, EventEmitter } from '@stencil/core';
+import { state, Screens } from '../../store';
+import { CloseIcon } from './assets/close-icon';
+import { ChevronIcon } from './assets/chevron-icon';
+import { Intro } from './screens/screens-intro';
+import { HowItWorks } from './screens/screens-how-it-works';
+import { AuthOptions } from '@blockstack/connect-core';
+
+@Component({
+  tag: 'connect-modal',
+  styleUrl: 'modal.sass',
+  assetsDirs: ['screens', 'assets'],
+})
+export class Modal {
+  @Prop() authOptions: AuthOptions;
+  @Event()
+  signUp: EventEmitter;
+
+  @Event()
+  signIn: EventEmitter;
```

Comment:
I miss describing events like this

Reply:
And it's all DOM events, which makes it easy to handle outside of the component. Stencil has a nice events API.